beacon_node/network/src/sync/message_processor.rs

```diff
@@ -447,24 +493,36 @@ impl<T: BeaconChainTypes> MessageProcessor<T> {
     /// Process a gossip message declaring a new attestation.
     ///
     /// Not currently implemented.
-    pub fn on_attestation_gossip(&mut self, _peer_id: PeerId, msg: Attestation<T::EthSpec>) {
+    pub fn on_attestation_gossip(&mut self, peer_id: PeerId, msg: Attestation<T::EthSpec>) {
         match self.chain.process_attestation(msg.clone()) {
-            Ok(outcome) => {
-                info!(
-                    self.log,
-                    "Processed attestation";
-                    "source" => "gossip",
-                    "outcome" => format!("{:?}", outcome)
-                );
-
-                if outcome != AttestationProcessingOutcome::Processed {
-                    trace!(
+            Ok(outcome) => match outcome {
+                AttestationProcessingOutcome::Processed => {
+                    info!(
                         self.log,
-                        "Invalid gossip attestation ssz";
-                        "ssz" => format!("0x{}", hex::encode(msg.as_ssz_bytes())),
+                        "Processed attestation";
+                        "source" => "gossip",
+                        "outcome" => format!("{:?}", outcome)
                     );
                 }
-            }
+                AttestationProcessingOutcome::UnknownHeadBlock { beacon_block_root } => {
+                    // TODO: Maintain this attestation and re-process once sync completes
+                    debug!(
+                    self.log,
+                    "Attestation for unknown block";
+                    "peer_id" => format!("{:?}", peer_id),
+                    "block" => format!("{}", beacon_block_root)
+                    );
+                    // we don't know the block, get the sync manager to handle the block lookup
+                    self.send_to_sync(SyncMessage::UnknownBlockHash(peer_id, beacon_block_root));
+                }
+                AttestationProcessingOutcome::AttestsToFutureState { .. }
+                | AttestationProcessingOutcome::FinalizedSlot { .. } => {} // ignore the attestation
+                AttestationProcessingOutcome::Invalid { .. }
+                | AttestationProcessingOutcome::EmptyAggregationBitfield { .. } => {
+                    // the peer has sent a bad attestation. Remove them.
+                    self.network.disconnect(peer_id, GoodbyeReason::Fault);
+                }
+            },
             Err(e) => {
                 trace!(
```

Comment:
Can we change this to an `error!`. We shouldn't get errors here and we want to know if it happens.

Reply:
Sure

Reply:
Yeah maybe 2 is good. The trace might be super verbose.

Reply:
This currently has a trace and error. Leaving as is